### PR TITLE
Improve UsernamePickerTestCase

### PR DESCRIPTION
**Pull request by richvdh**
* make the OIDC bits of the test work at a higher level - via the REST api instead of poking the OIDCHandler directly.
* Move it to `test_login.py`, where I think it fits better.

### CODE REVIEW

File: tests/handlers/test_oidc.py [hunk 1]
```diff
@@ -855,69 +854,60 @@ def _generate_oidc_session_token(
         )
 
 
+@skip_unless(HAS_OIDC, "requires OIDC")
```
Comment (clokep) [inline tests/handlers/test_oidc.py]: I saw we made this change in other places too -- is there a benefit to `skip_unless`? Or just a bit simpler?

File: tests/handlers/test_oidc.py [hunk 2]
```diff
@@ -855,69 +854,60 @@ def _generate_oidc_session_token(
...
-                "config": {"display_name_template": "{{ user.displayname }}"}
-            },
-        }
 
-        # Update this config with what's in the default config so that
-        # override_config works as expected.
-        oidc_config.update(config.get("oidc_config", {}))
-        config["oidc_config"] = oidc_config
+        config["oidc_config"] = {}
+        config["oidc_config"].update(TEST_OIDC_CONFIG)
```
Comment (clokep) [inline tests/handlers/test_oidc.py]: Is this just to make a copy? Would it be clearer to do `config["oidc_config"] = TEST_OIDC_CONFIG.copy()`?

**Review by clokep — APPROVED** [review]
Looks like a nice simplification!

File: tests/handlers/test_oidc.py (hunk 1 above)
Comment (richvdh) [reply tests/handlers/test_oidc.py]: just simpler.

File: tests/handlers/test_oidc.py (hunk 2 above)
Comment (richvdh) [reply tests/handlers/test_oidc.py]: I guess... I tend to forget about the existence of `dict.copy` :/

**Pull request merged**